API/AddUser.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+    ini_set('display_errors', 1);
+    ini_set('error_reporting', E_ALL);
+
+	$inData = getRequestInfo();
+
+	$servername = "localhost";
+	$sv_username = "root";
+	$sv_password = "";
+	$dbname = "myDB";
+
+	// Create connection
+	$conn = new mysqli($servername, $sv_username, $sv_password, $dbname); 	
+	if( $conn->connect_error )
+	{
+		returnWithError( $conn->connect_error );
+	}
+	else
+	{
+		if (userExists($conn, $inData['username']) != false)
+		{
+			returnWithError("Username Already Taken");
+		}
+		else
+		{
+			$hashed_password = password_hash($inData["password"], PASSWORD_DEFAULT);
+			
+			$stmt = $conn->prepare("INSERT INTO Users (ID, firstName, lastName, username, password, dateCreated) VALUES (DEFAULT, ?, ?, ?, ?, DEFAULT)");
+			$stmt->bind_param("ssss", $inData["firstName"], $inData["lastName"], $inData["username"], $hashed_password);
+			$stmt->execute();
+			$result = $stmt->get_result();
+			$row = $result->fetch_assoc();
+			returnWithInfo($row['firstName'], $row['lastName'], $row['ID']);
```

Comment:
Since we can't call fetch_assoc we can't really use $row. One of the possible solutions would be to make a SELECT FROM query which Username that we used during creation. In this case we will be able to get all "auto generated" info about new user.